### ISSUE TITLE
NYU test results not good as it should be

### ISSUE THREAD

**Issue by MeteorKepler** [melonwan/denseReg#8]
After download nyu dataset and your pretrained model, I run the command `python model/hourglass_um_crop_tiny.py --dataset 'nyu' --batch_size 3 --num_stack 2 --num_fea 128 --debug_level 2` with is_train equals False in the code. But the results is not good as it should be:
```
[2018-04-19 16:06:11.933652]: 2727/393226 computed, with 0.10s
finish test
10mm percentage: 0.012239
20mm percentage: 0.057683
30mm percentage: 0.072225
40mm percentage: 0.095613
```
This is what i see in tensorboard:

Ground Truth:
![snipaste_2018-04-19_16-14-50](https://user-images.githubusercontent.com/18494974/38979672-690e8c9a-43ed-11e8-8198-ca88c5717eba.png)
Val:
![snipaste_2018-04-19_16-15-07](https://user-images.githubusercontent.com/18494974/38979680-6d061854-43ed-11e8-9d5b-3b7cb8d831e7.png)

And this is the generating result files:
[testing-2018-04-19_15_59_22.194820-result.txt](https://github.com/melonwan/denseReg/files/1927204/testing-2018-04-19_15_59_22.194820-result.txt)
[testing-2018-04-19_15_59_22.194820-result_error.txt](https://github.com/melonwan/denseReg/files/1927205/testing-2018-04-19_15_59_22.194820-result_error.txt)

What's the problem?

My environment:
- Ubuntu 16.04
- GTX titan xp 12GB
- cuda9.0 + cudnn7
- python 2.7
- tensorflow 1.7
- tfplot 0.2.0
- matplotlib 2.2.2
- opencv 2.4.11

**Comment by melonwan** [comment]
Hi, 
   I've downloaded the code and model, run again and the result is as expected. According to the tensorboard visualization, there could be something wrong with the preprocessing step, and some code has been changed.

**Comment by melonwan** [comment]
No, the original code should run correctly. Sorry for my ambiguious explaination, I just mean by some original code might be changed on your side which makes the difference.

**Comment by MeteorKepler** [comment]
I cloned your code again and run from begining, but it came to the same results. I think you should solve the issue  #4 so that we can evaluate dataset icvl and msra to see if it is good for them

**Comment by MeteorKepler** [comment]
Yes, i run it several time and made sure the code I modified does nothing to do with data and network. Maybe it's my environment or dataset corruption. So i will use another PC to rerun and check other two dataset.

**Comment by MeteorKepler** [comment]
I run the code with ICVL and the result is normal as expected. 
I don't know why it's strange on NYU. Maybe the pretrained model you offer is somewhat inaccurate. Could you please check and reupload most recent checkpoint trained on NYU?
Thanks a lot for your effort!

Another code modification suggestion:
delete line 51 in data/icvl.py `return True` in `is_train()` or it will not load test data

**Comment by MeteorKepler** [comment]
I got it. Your NYU test dataset is not the same as NYU on official website now. The dataset has updated. But the nyu_bbx file you offer remains unchanged. So many samples don't crop the hand location.

**Comment by melonwan** [comment]
Hi, thank you for telling me that. Actually, this is not a "bug", but an implementation trick. We only estimated the bounding box of view 1, which is also the view to do evaluation, and set the rest view with the same bounding box, which is not the correct one, but can obtain the same result. I've run on my side with exactly the same code online.

**Comment by melonwan** [comment]
Hey, 'view' means camera view. Yes, the code only evaluate view 1, as this is the standard evaluation protocol.

**Comment by MeteorKepler** [comment]
So, you only take bbox frame 1 of view 1 to do all frames cropping of that view in your code. That's what i think inaccurate.

**Comment by melonwan** [comment]
no, it is bbx of view 1, preframe, corresponds to view1, 2, 3 bbx, which is wrong for view 2, 3, but works correctly on view1, which is the evaluation view.

**Comment by MeteorKepler** [comment]
I mean in your original code
```
for c_j, c_n, c_b in zip(joints, names, bbxes):
    for j, n in zip(c_j, c_n):
```
len(joints)=1, len(names)=1, len(bbxes)=8252.
If using zip(), you will not get bbxes of frame 2-8252.
So i change your code as mentioned above.

**Comment by melonwan** [comment]
Thanks a lot for informing me about this. I'll update accordingly.

**Comment by MeteorKepler** [comment]
Have you ever tried RGB-D input? I mean changing input of [B, H, W, 1] to [B, H, W, 4] and train your model.
Of course it will increase computation, but will this increase the accuracy?

**Comment by melonwan** [comment]
This idea is great. Actually I've tried with the similar idea, to use hand color to crop out hand. Unfortunately, rgb and d is not calibrated and well-aligned for the nyu dataset.
